Rename references to "captive portal mode" in WebKit to "Lockdown mode"

whsieh opened the pull request:
#### 8668ed77138a6e98d69d0cb8854b1c2254d54a17
<pre>
Rename references to &quot;captive portal mode&quot; in WebKit to &quot;Lockdown mode&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=246648">https://bugs.webkit.org/show_bug.cgi?id=246648</a>

Reviewed by Geoffrey Garen and Chris Dumez.

Mechanically rename all &quot;captive portal&quot; code in WebKit to Lockdown mode instead, to reflect the
true purpose of this code. Note that we avoid renaming:

- Exported SPI or API methods.
- The child process name (&quot;com.apple.WebKit.WebContent.CaptivePortal&quot;)

...to avoid breaking any clients (Apple-internal or third party) that may be relying on these
existing names.

* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::maybeLoadFallbackSynchronously):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp:
(WebKit::GPUConnectionToWebProcess::createGraphicsContextGL):
(WebKit::GPUConnectionToWebProcess::releaseGraphicsContextGL):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h:
(WebKit::GPUConnectionToWebProcess::isLockdownModeEnabled const):
(WebKit::GPUConnectionToWebProcess::isCaptivePortalModeEnabled const): Deleted.
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp:
(WebKit::RemoteAudioDestinationManager::createAudioDestination):
(WebKit::RemoteAudioDestinationManager::deleteAudioDestination):
(WebKit::RemoteAudioDestinationManager::startAudioDestination):
(WebKit::RemoteAudioDestinationManager::stopAudioDestination):
* Source/WebKit/Shared/GPUProcessConnectionParameters.h:
(WebKit::GPUProcessConnectionParameters::encode const):
(WebKit::GPUProcessConnectionParameters::decode):
* Source/WebKit/Shared/WebProcessCreationParameters.cpp:
(WebKit::WebProcessCreationParameters::encode const):
(WebKit::WebProcessCreationParameters::decode):
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::lockdownModeEnabled const):
(API::PageConfiguration::isLockdownModeExplicitlySet const):
(API::PageConfiguration::captivePortalModeEnabled const): Deleted.
(API::PageConfiguration::isCaptivePortalModeExplicitlySet const): Deleted.
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
(API::WebsitePolicies::lockdownModeEnabled const):
(API::WebsitePolicies::captivePortalModeEnabled const): Deleted.
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessPool.mm:
(+[WKProcessPool _setCaptivePortalModeEnabledGloballyForTesting:]):
(+[WKProcessPool _lockdownModeEnabledGloballyForTesting]):
(+[WKProcessPool _clearCaptivePortalModeEnabledGloballyForTesting]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences init]):
(-[WKWebpagePreferences _setCaptivePortalModeEnabled:]):
(-[WKWebpagePreferences _captivePortalModeEnabled]):
(-[WKWebpagePreferences isLockdownModeEnabled]):
(-[WKWebpagePreferences setLockdownModeEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKSystemPreferences.mm:
(+[_WKSystemPreferences isCaptivePortalModeEnabled]):
(+[_WKSystemPreferences setCaptivePortalModeEnabled:]):
(+[_WKSystemPreferences setCaptivePortalModeIgnored:ignore:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKSystemPreferencesInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView didMoveToWindow]):
(isLockdownModeWarningNeeded):
(-[WKWebView _presentLockdownMode]):
(-[WKWebView _presentLockdownModeAlertIfNeeded]):
(-[WKWebView _presentCaptivePortalMode]): Deleted.
(-[WKWebView _presentCaptivePortalModeAlertIfNeeded]): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::cachedLockdownModeEnabledGlobally):
(WebKit::WebProcessPool::lockdownModeConfigUpdateCallback):
(WebKit::WebProcessPool::registerNotificationObservers):
(WebKit::WebProcessPool::unregisterNotificationObservers):
(WebKit::lockdownModeObservers):
(WebKit::isLockdownModeEnabledGloballyForTesting):
(WebKit::isLockdownModeEnabledBySystemIgnoringCaching):
(WebKit::WebProcessPool::lockdownModeStateChanged):
(WebKit::addLockdownModeObserver):
(WebKit::removeLockdownModeObserver):
(WebKit::lockdownModeEnabledBySystem):
(WebKit::setLockdownModeEnabledGloballyForTesting):
(WebKit::WebProcessPool::notifyPreferencesChanged):
(WebKit::cachedCaptivePortalModeEnabledGlobally): Deleted.
(WebKit::WebProcessPool::captivePortalModeConfigUpdateCallback): Deleted.
(WebKit::captivePortalModeObservers): Deleted.
(WebKit::isCaptivePortalModeEnabledGloballyForTesting): Deleted.
(WebKit::isCaptivePortalModeEnabledBySystemIgnoringCaching): Deleted.
(WebKit::WebProcessPool::captivePortalModeStateChanged): Deleted.
(WebKit::addCaptivePortalModeObserver): Deleted.
(WebKit::removeCaptivePortalModeObserver): Deleted.
(WebKit::captivePortalModeEnabledBySystem): Deleted.
(WebKit::setCaptivePortalModeEnabledGloballyForTesting): Deleted.
* Source/WebKit/UIProcess/Launcher/ProcessLauncher.h:
(WebKit::ProcessLauncher::Client::shouldEnableLockdownMode const):
(WebKit::ProcessLauncher::Client::shouldEnableCaptivePortalMode const): Deleted.
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::webContentServiceName):
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/LockdownModeObserver.h: Renamed from Source/WebKit/UIProcess/CaptivePortalModeObserver.h.
(WebKit::LockdownModeObserver::~LockdownModeObserver):
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::SuspendedPageProxy::findReusableSuspendedPageProcess):
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::receivedNavigationPolicyDecision):
(WebKit::WebPageProxy::decidePolicyForResponseShared):
(WebKit::WebPageProxy::triggerBrowsingContextGroupSwitchForNavigation):
(WebKit::WebPageProxy::shouldEnableLockdownMode const):
(WebKit::WebPageProxy::shouldEnableCaptivePortalMode const): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::takeProcess):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::createGPUProcessConnection):
(WebKit::WebProcessPool::createNewWebProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::initializeNewWebProcess):
(WebKit::WebProcessPool::prewarmProcess):
(WebKit::WebProcessPool::processForRegistrableDomain):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
(WebKit::addLockdownModeObserver):
(WebKit::removeLockdownModeObserver):
(WebKit::lockdownModeEnabledBySystem):
(WebKit::setLockdownModeEnabledGloballyForTesting):
(WebKit::addCaptivePortalModeObserver): Deleted.
(WebKit::removeCaptivePortalModeObserver): Deleted.
(WebKit::captivePortalModeEnabledBySystem): Deleted.
(WebKit::setCaptivePortalModeEnabledGloballyForTesting): Deleted.
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::WebProcessProxy::WebProcessProxy):
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::lockdownMode const):
(WebKit::WebProcessProxy::captivePortalMode const): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::adjustSettingsForLockdownMode):
(WebKit::WebPage::updatePreferences):
(adjustCoreGraphicsForCaptivePortal): Deleted.

Remove this helper method, and move the function calls to the call site in
`adjustSettingsForLockdownMode`. Additionally add a FIXME, regarding the placement of these function
calls.

(WebKit::adjustSettingsForCaptivePortal): Deleted.
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeWebProcess):
* Source/WebKit/WebProcess/WebProcess.h:
(WebKit::WebProcess::isLockdownModeEnabled const):
(WebKit::WebProcess::isCaptivePortalModeEnabled const): Deleted.
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CaptivePortalPDF.html:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/LockdownModeFonts.mm: Renamed from Tools/TestWebKitAPI/Tests/WebKitCocoa/CaptivePortalModeFonts.mm.
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/MediaLoading.mm:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm:
(-[LockdownModeKVO observeValueForKeyPath:ofObject:change:context:]):
(-[CaptivePortalModeKVO observeValueForKeyPath:ofObject:change:context:]): Deleted.
(-[CaptivePortalMessageHandler userContentController:didReceiveScriptMessage:]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:

Canonical link: <a href="https://commits.webkit.org/255664@main">https://commits.webkit.org/255664@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7c32289e895a7a4a7b42a7fa8bc0a736b8bd8343

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93200 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2399 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23851 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102905 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/163193 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2405 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30721 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85591 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99018 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98869 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/1670 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79669 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28589 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/83583 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/83349 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/71699 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/84520 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37112 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17228 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/79580 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34929 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18470 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/27526 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3925 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38800 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41004 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/82213 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40730 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37680 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/18586 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
```diff
@@ -280,7 +280,7 @@ bool ApplicationCacheHost::maybeLoadSynchronously(ResourceRequest& request, Reso
 
 void ApplicationCacheHost::maybeLoadFallbackSynchronously(const ResourceRequest& request, ResourceError& error, ResourceResponse& response, RefPtr<SharedBuffer>& data)
 {
-    // If normal loading results in a redirect to a resource with another origin (indicative of a captive portal), or a 4xx or 5xx status code or equivalent,
```
I don't think this rename is correct. This really meant captive portal :)

whsieh replied:
Fixed!

geoffreygaren reviewed (approved):
r=me

pxlcoder commented on Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
```diff
@@ -280,7 +280,7 @@ bool ApplicationCacheHost::maybeLoadSynchronously(ResourceRequest& request, Reso
 void ApplicationCacheHost::maybeLoadFallbackSynchronously(const ResourceRequest& request, ResourceError& error, ResourceResponse& response, RefPtr<SharedBuffer>& data)
 {
-    // If normal loading results in a redirect to a resource with another origin (indicative of a captive portal), or a 4xx or 5xx status code or equivalent,
+    // If normal loading results in a redirect to a resource with another origin (indicative of a Lockdown), or a 4xx or 5xx status code or equivalent,
```
Probably want to undo this one?

whsieh replied:
Whoops, good catch! Looks this one is really about captive portals 😓.

cdumez reviewed (approved):
LGTM with minor change.

whsieh committed:
Rename references to "captive portal mode" in WebKit to "Lockdown mode"
https://bugs.webkit.org/show_bug.cgi?id=246648

Reviewed by Geoffrey Garen and Chris Dumez.

Mechanically rename all "captive portal" code in WebKit to Lockdown mode instead, to reflect the
true purpose of this code. Note that we avoid renaming:

- Exported SPI or API methods.
- The child process name ("com.apple.WebKit.WebContent.CaptivePortal")

...to avoid breaking any clients (Apple-internal or third party) that may be relying on these
existing names.

* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
(WebCore::ApplicationCacheHost::maybeLoadFallbackSynchronously):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp:
(WebKit::GPUConnectionToWebProcess::createGraphicsContextGL):
(WebKit::GPUConnectionToWebProcess::releaseGraphicsContextGL):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h:
(WebKit::GPUConnectionToWebProcess::isLockdownModeEnabled const):
(WebKit::GPUConnectionToWebProcess::isCaptivePortalModeEnabled const): Deleted.
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp:
(WebKit::RemoteAudioDestinationManager::createAudioDestination):
(WebKit::RemoteAudioDestinationManager::deleteAudioDestination):
(WebKit::RemoteAudioDestinationManager::startAudioDestination):
(WebKit::RemoteAudioDestinationManager::stopAudioDestination):
* Source/WebKit/Shared/GPUProcessConnectionParameters.h:
(WebKit::GPUProcessConnectionParameters::encode const):
(WebKit::GPUProcessConnectionParameters::decode):
* Source/WebKit/Shared/WebProcessCreationParameters.cpp:
(WebKit::WebProcessCreationParameters::encode const):
(WebKit::WebProcessCreationParameters::decode):
* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/UIProcess/API/APIPageConfiguration.cpp:
(API::PageConfiguration::lockdownModeEnabled const):
(API::PageConfiguration::isLockdownModeExplicitlySet const):
(API::PageConfiguration::captivePortalModeEnabled const): Deleted.
(API::PageConfiguration::isCaptivePortalModeExplicitlySet const): Deleted.
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
(API::WebsitePolicies::lockdownModeEnabled const):
(API::WebsitePolicies::captivePortalModeEnabled const): Deleted.
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKProcessPool.mm:
(+[WKProcessPool _setCaptivePortalModeEnabledGloballyForTesting:]):
(+[WKProcessPool _lockdownModeEnabledGloballyForTesting]):
(+[WKProcessPool _clearCaptivePortalModeEnabledGloballyForTesting]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences init]):
(-[WKWebpagePreferences _setCaptivePortalModeEnabled:]):
(-[WKWebpagePreferences _captivePortalModeEnabled]):
(-[WKWebpagePreferences isLockdownModeEnabled]):
(-[WKWebpagePreferences setLockdownModeEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKSystemPreferences.mm:
(+[_WKSystemPreferences isCaptivePortalModeEnabled]):
(+[_WKSystemPreferences setCaptivePortalModeEnabled:]):
(+[_WKSystemPreferences setCaptivePortalModeIgnored:ignore:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKSystemPreferencesInternal.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView didMoveToWindow]):
(isLockdownModeWarningNeeded):
(-[WKWebView _presentLockdownMode]):
(-[WKWebView _presentLockdownModeAlertIfNeeded]):
(-[WKWebView _presentCaptivePortalMode]): Deleted.
(-[WKWebView _presentCaptivePortalModeAlertIfNeeded]): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::cachedLockdownModeEnabledGlobally):
(WebKit::WebProcessPool::lockdownModeConfigUpdateCallback):
(WebKit::WebProcessPool::registerNotificationObservers):
(WebKit::WebProcessPool::unregisterNotificationObservers):
(WebKit::lockdownModeObservers):
(WebKit::isLockdownModeEnabledGloballyForTesting):
(WebKit::isLockdownModeEnabledBySystemIgnoringCaching):
(WebKit::WebProcessPool::lockdownModeStateChanged):
(WebKit::addLockdownModeObserver):
(WebKit::removeLockdownModeObserver):
(WebKit::lockdownModeEnabledBySystem):
(WebKit::setLockdownModeEnabledGloballyForTesting):
(WebKit::WebProcessPool::notifyPreferencesChanged):
(WebKit::cachedCaptivePortalModeEnabledGlobally): Deleted.
(WebKit::WebProcessPool::captivePortalModeConfigUpdateCallback): Deleted.
(WebKit::captivePortalModeObservers): Deleted.
(WebKit::isCaptivePortalModeEnabledGloballyForTesting): Deleted.
(WebKit::isCaptivePortalModeEnabledBySystemIgnoringCaching): Deleted.
(WebKit::WebProcessPool::captivePortalModeStateChanged): Deleted.
(WebKit::addCaptivePortalModeObserver): Deleted.
(WebKit::removeCaptivePortalModeObserver): Deleted.
(WebKit::captivePortalModeEnabledBySystem): Deleted.
(WebKit::setCaptivePortalModeEnabledGloballyForTesting): Deleted.
* Source/WebKit/UIProcess/Launcher/ProcessLauncher.h:
(WebKit::ProcessLauncher::Client::shouldEnableLockdownMode const):
(WebKit::ProcessLauncher::Client::shouldEnableCaptivePortalMode const): Deleted.
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::webContentServiceName):
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/LockdownModeObserver.h: Renamed from Source/WebKit/UIProcess/CaptivePortalModeObserver.h.
(WebKit::LockdownModeObserver::~LockdownModeObserver):
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::SuspendedPageProxy::findReusableSuspendedPageProcess):
* Source/WebKit/UIProcess/SuspendedPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::receivedNavigationPolicyDecision):
(WebKit::WebPageProxy::decidePolicyForResponseShared):
(WebKit::WebPageProxy::triggerBrowsingContextGroupSwitchForNavigation):
(WebKit::WebPageProxy::shouldEnableLockdownMode const):
(WebKit::WebPageProxy::shouldEnableCaptivePortalMode const): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebProcessCache.cpp:
(WebKit::WebProcessCache::takeProcess):
* Source/WebKit/UIProcess/WebProcessCache.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::createGPUProcessConnection):
(WebKit::WebProcessPool::createNewWebProcess):
(WebKit::WebProcessPool::tryTakePrewarmedProcess):
(WebKit::WebProcessPool::initializeNewWebProcess):
(WebKit::WebProcessPool::prewarmProcess):
(WebKit::WebProcessPool::processForRegistrableDomain):
(WebKit::WebProcessPool::createWebPage):
(WebKit::WebProcessPool::processForNavigation):
(WebKit::WebProcessPool::processForNavigationInternal):
(WebKit::addLockdownModeObserver):
(WebKit::removeLockdownModeObserver):
(WebKit::lockdownModeEnabledBySystem):
(WebKit::setLockdownModeEnabledGloballyForTesting):
(WebKit::addCaptivePortalModeObserver): Deleted.
(WebKit::removeCaptivePortalModeObserver): Deleted.
(WebKit::captivePortalModeEnabledBySystem): Deleted.
(WebKit::setCaptivePortalModeEnabledGloballyForTesting): Deleted.
* Source/WebKit/UIProcess/WebProcessPool.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::create):
(WebKit::WebProcessProxy::createForRemoteWorkers):
(WebKit::WebProcessProxy::WebProcessProxy):
* Source/WebKit/UIProcess/WebProcessProxy.h:
(WebKit::WebProcessProxy::lockdownMode const):
(WebKit::WebProcessProxy::captivePortalMode const): Deleted.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::adjustSettingsForLockdownMode):
(WebKit::WebPage::updatePreferences):
(adjustCoreGraphicsForCaptivePortal): Deleted.

Remove this helper method, and move the function calls to the call site in
`adjustSettingsForLockdownMode`. Additionally add a FIXME, regarding the placement of these function
calls.

(WebKit::adjustSettingsForCaptivePortal): Deleted.
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeWebProcess):
* Source/WebKit/WebProcess/WebProcess.h:
(WebKit::WebProcess::isLockdownModeEnabled const):
(WebKit::WebProcess::isCaptivePortalModeEnabled const): Deleted.
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CaptivePortalPDF.html:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/LockdownModeFonts.mm: Renamed from Tools/TestWebKitAPI/Tests/WebKitCocoa/CaptivePortalModeFonts.mm.
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/MediaLoading.mm:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm:
(-[LockdownModeKVO observeValueForKeyPath:ofObject:change:context:]):
(-[CaptivePortalModeKVO observeValueForKeyPath:ofObject:change:context:]): Deleted.
(-[CaptivePortalMessageHandler userContentController:didReceiveScriptMessage:]): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:

Canonical link: https://commits.webkit.org/255664@main
